source-code/cli/src/commands/machine/translate.ts

```diff
@@ -0,0 +1,168 @@
+	log.info(
+		"🌏 Found " +
+			Object.keys(messageCounts).length +
+			" languages with a total of " +
+			Object.values(messageCounts).reduce((a, b) => a + b, 0) +
+			" messages.",
+	)
+
+	// Get languages to translate to with the reference language removed
+	const languagesToTranslateTo = resources.filter(
+		(resource) => resource.languageTag.name !== referenceLanguage,
+	)
+	log.info(
+		"📝 Translating to " +
+			languagesToTranslateTo.length +
+			" languages. [" +
+			[...new Set(languagesToTranslateTo)]
+				.splice(0, 3)
+				.map((language) => language.languageTag.name)
+				.join(", ") +
+			"]",
+	)
+
+	// Translate all messages
+	for (const language of languagesToTranslateTo) {
+		for (const message of referenceLanguageResource.body) {
+			// skip if message already exists in language
+			if (language.body.some((langMessage) => langMessage.id.name === message.id.name)) {
+				continue
+			}
+
+			// 🌏 Translation
+			const [translation, exception] = await rpc.machineTranslate({
+				telemetryId: "CLI",
+				referenceLanguage: referenceLanguage,
+				targetLanguage: language.languageTag.name,
+				text: message.pattern.elements[0]!.value as string,
+			})
+			if (exception) {
+				log.error("Couldn't translate message " + message.id.name + ". ", exception.message)
+				continue
+			}
+
+			const bold = (text: string) => `\x1b[1m${text}\x1b[0m`
+			const italic = (text: string) => `\x1b[3m${text}\x1b[0m`
+			log.info(
+				getFlag(language.languageTag.name) +
+					" Translated message " +
+					bold(message.id.name) +
+					" to " +
+					italic(translation),
+			)
+
+			const newMessage: Message = {
+				type: "Message",
+				id: { type: "Identifier", name: message.id.name },
+				pattern: {
+					type: "Pattern",
+					elements: [{ type: "Text", value: translation }],
+				},
+			}
+
+			// find language resource to add the new message to
+			const languageResource = resources.find(
+				(resource) => resource.languageTag.name === language.languageTag.name,
+			)
+			if (languageResource) {
+				const [newResource, exception] = query(languageResource).upsert({
+					message: newMessage,
+				})
+
+				if (exception) {
+					log.error("Couldn't upsert new message. ", exception.message)
+				}
+
+				// merge the new resource with the existing resources
+				resources = resources.map((resource) => {
+					if (resource.languageTag.name === language.languageTag.name && newResource) {
+						return newResource
+					}
+					return resource
+				})
+			}
+		}
+	}
+
+	// write the new resource to the file system
+	await config.writeResources({
+		config,
+		resources: resources,
+	})
+
+	// get remote origin url by executing git command
+	exec("git config --get remote.origin.url", (error, stdout) => {
+		if (error) {
+			console.error("Failed to get the remote URL.", error.message)
+			return
+		}
+
+		const remoteUrl = stdout.trim()
+
+		telemetryNode.capture({
+			distinctId: "CLI",
+			event: "CLI: machine translate",
```

Comment:
Let's name it `CLI command executed [machine translate]` to have a common way of identifying that a command has been executed?

Reply:
done